src/nvim/event/socket.c

```diff
@@ -40,17 +40,18 @@ int socket_watcher_init(Loop *loop, SocketWatcher *watcher,
       return UV_EINVAL;
     }
 
-    *host_end = '\0';
     uv_getaddrinfo_t request;
+    *host_end = '\0';
+    char *host = !strcmp(addr, "localhost") ? "127.0.0.1" : addr;
```

Comment:
What if `localhost` is defined and its IP address is not `127.0.0.1` (perhaps a system with only IPv6 addresses that  maps `localhost` to `::1`? 
Also the the name in `serverlist()` would not match the address that the socket is using. Should names in the list be replaced with the IP address that is used with the tcp socket?